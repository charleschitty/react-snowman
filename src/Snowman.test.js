import {render, fireEvent} from "@testing-library/react";
import Snowman from "./Snowman";
import TEST_IMAGES from "./_testCommon";
import React, {useState} from "react";


test("renders without crashing", function () {
  render(<Snowman />);
})

test("if game properly ends when losing", function (){


  const testWord = "apple"

  const{ container, debug } = render(
    <Snowman images={TEST_IMAGES} words={testWord} maxWrong={TEST_IMAGES.length} />
  );

  const zButton = container.querySelector('.Snowman button[value="z"]');
  fireEvent.click(zButton);

  const xButton = container.querySelector('.Snowman button[value="x"]');
  fireEvent.click(xButton);

  const gButton = container.querySelector('.Snowman button[value="g"]');
  fireEvent.click(gButton);

  const yButton = container.querySelector('.Snowman button[value="y"]');
  fireEvent.click(yButton);

  const kButton = container.querySelector('.Snowman button[value="k"]');
  fireEvent.click(kButton);

  const vButton = container.querySelector('.Snowman button[value="v"]');
  fireEvent.click(vButton);

  // Expect the buttons to no longer appear on page
  expect(
    container.querySelector('.Snowman button')
  ).not.toBeInTheDocument();

  // Expect to see the last snowman image on page
  expect(
    container.querySelector('img[alt="testing image 6"]')
  ).toBeInTheDocument();

  // Expect to see the end game message
  expect(
    container.querySelector('.end-game-message')
  ).toHaveTextContent('You lose');

  // Expect to see the target word
  expect(
    container.querySelector('.end-game-message')
  ).toHaveTextContent('apple');

});
