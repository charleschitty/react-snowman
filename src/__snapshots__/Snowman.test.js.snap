// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if game properly ends when losing 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="6"
    />
    <p>
      Number wrong: 
      6
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p>
      <p
        class="Snowman-end-message"
      >
        You lose. Word was 
        apple
        .
      </p>
    </p>
  </div>
</div>
`;
